### PR TITLE
Bug Fix: Far Camera Blending Not Working

### LINKED ISSUE #80: Camera pan out upon grabbing dentures doesn't zoom out/move correctly
The camera moves back as if it's going to show the responder AI, but it doesn't zoom out to show the player anything.

### PR DESCRIPTION
**Pull request by jurcaua**
Fixes the issue Finnbarr logged: 
Fixes #80 

TLDR; Awareness affecting camera distance also affected the far camera, so I worked on keeping the internal state of `canZoom` in `CameraManager` and this toggles off/on during the far camera blend to and from. 

---

This fix needed a good amount of abstraction in `CameraManager` to keep it at least somewhat non-hardcoded.
* Main idea is that for some blending operations, like blending to the far camera for some time, do not need to alert listeners like, the player which will disable movement and stuff during the camera movement.
  * We have two event types currently, `OnBlendingStart` and `OnBlendingComplete`
* But internally, we need to listen to these events to now stop the ability for outside sources to affect things like, zoom level, which was the main issue for this bug.
* So now we have internal events, which are private, in addition to the public-facing events we already have.
* Now when we want to blend cameras, we can stick with the default public events, or pass in our own internal events and listen to those, which are guaranteed to not be listened to by any outside sources.

### CODE REVIEW

File: Assets/Scripts/CameraManager.cs [hunk 1]
```diff
@@ -153,19 +168,32 @@ public IEnumerator BlendToFarCameraForSeconds(float seconds)
...
+        OnBlendingComplete_Internal -= ReenableZoomAfterBlend;
+        canZoom = true;
+    }
 
-        // Update the our local activeCameraDefaultCameraDistance
-        CinemachineFramingTransposer blendToCameraFramingTransposer = GetActiveVirtualCamera().GetCinemachineComponent<CinemachineFramingTransposer>();
-        if (blendToCameraFramingTransposer)
+    private void UpdateDefaultCameraDistance(CinemachineVirtualCamera virtualCamera)
+    {
+        if (canZoom)
```
Comment (fnnbrr) [inline Assets/Scripts/CameraManager.cs:187]: Works fine for now & is clear to understand.

We can think about ways to improve this in the future; could a single variable (e.g. "playerLocked") be used to indicate that A) the player can't move, and B) nothing should be modifying the camera besides internal functions?  It could be stored in GameManager or PlayerManager, and we'll probably end up using it for other things as well.

Comment (fnnbrr) [reply Assets/Scripts/CameraManager.cs:187]: Ideally we can also move this "if (canZoom)" check further up (into "BlendTo"?) to prevent running through unnecessary lines of code.

**Review by fnnbrr — APPROVED** [review]
Works for me, and we can refactor later on if we need to implement more systems that rely on the player being "locked" in place w/o inputs to movement or camera from external sources.

File: Assets/Scripts/CameraManager.cs (hunk 1 above)
Comment (jurcaua) [reply Assets/Scripts/CameraManager.cs:187]: Thanks! 😄 

Well currently in `PlayerController` we have a public bool: `CanMove`, so I'm guessing that's the kind of thing you're referring to? Or are you be looking to add further functionality?

Also the "zoom" and the camera blending are separate things (as in, "zooming" only affects the current camera and how much farther/less it is zoomed in compared to it's original camera distance.
The blending is switching between two cameras, and the main issue was having the `AwarenessManager` constantly updating the "zoom" of the close camera, but then when blending to the far camera, we wanted no zoom to occur and just keep it's originally set camera distance.

So I'm not sure we can move the `if (canZoom)` check outside of this function? From what I can see.

Thanks for the quick review sir!
